src/components/WebPlatformPlayer/index.js

```diff
@@ -0,0 +1,445 @@
+          screen: ScreenName.SignTransactionSummary,
+          params: {
+            currentNavigation: ScreenName.SignTransactionSummary,
+            nextNavigation: ScreenName.SignTransactionSelectDevice,
+            transaction: tx,
+            accountId,
+            appName: params.useApp,
+            onSuccess: ({ signedOperation, transactionSignError }) => {
+              if (transactionSignError) reject(transactionSignError);
+              else {
+                resolve(signedOperation);
+                const n = navigation.dangerouslyGetParent() || navigation;
+                n.dangerouslyGetParent().pop();
+              }
+            },
+            onError: reject,
+          },
+        });
+      });
+    },
+    [accounts, navigation],
+  );
+
+  const broadcastTransaction = useCallback(
+    async ({
+      accountId,
+      signedTransaction,
+    }: {
+      accountId: string,
+      signedTransaction: SignedOperation,
+    }) => {
+      const account = accounts.find(a => a.id === accountId);
+
+      return new Promise((resolve, reject) => {
+        // @TODO replace with correct error
+        if (!signedTransaction) {
+          reject(new Error("Transaction required"));
+          return;
+        }
+        if (!account) {
+          reject(new Error("Account required"));
+          return;
+        }
+
+        if (!getEnv("DISABLE_TRANSACTION_BROADCAST")) {
+          broadcastSignedTx(account, null, signedTransaction).then(
+            op => resolve(op.hash),
+            reject,
+          );
+        }
+      });
+    },
+    [accounts],
+  );
+
+  const handlers = useMemo(
+    () => ({
+      "account.list": listAccounts,
+      "currency.list": listCurrencies,
+      "account.request": requestAccount,
+      "account.receive": receiveOnAccount,
+      "transaction.sign": signTransaction,
+      "transaction.broadcast": broadcastTransaction,
+    }),
+    [
+      listAccounts,
+      listCurrencies,
+      requestAccount,
+      receiveOnAccount,
+      signTransaction,
+      broadcastTransaction,
+    ],
+  );
+
+  const handleSend = useCallback(
+    (request: JSONRPCRequest) => {
+      targetRef?.current?.postMessage(
+        JSON.stringify(request),
+        manifest.url.origin,
+      );
+    },
+    [manifest],
+  );
+
+  const [receive] = useJSONRPCServer(handlers, handleSend);
+
+  const handleMessage = useCallback(
+    e => {
+      // FIXME: event isn't the same on desktop & mobile
+      // if (e.isTrusted && e.origin === manifest.url.origin && e.data) {
+      if (e.nativeEvent?.data) {
+        receive(JSON.parse(e.nativeEvent.data));
+      }
+    },
+    [receive],
+  );
+
+  const handleLoad = useCallback(() => {
+    setWidgetError(false);
+    setWidgetLoaded(true);
+  }, []);
+
+  const handleReload = useCallback(() => {
+    setLoadDate(Date.now());
+    setWidgetLoaded(false);
+  }, []);
+
+  React.useLayoutEffect(() => {
+    navigation.setOptions({
```

Comment:
please assess that doing this is not heavy (does it re-render this component itself? no infinite loop?)

Reply:
So apparently we render the full component 3 times at each (re)loading. No idea why... but I checked that this part isn't responsible of it, it doesn't add any re-rendering.